ml/Host.cc

```diff
@@ -0,0 +1,242 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+#include "Config.h"
+#include "Host.h"
+
+#include "json/single_include/nlohmann/json.hpp"
+
+using namespace ml;
+
+static void updateMLChart(RRDHOST *RH,
+                          collected_number NumTotalDimensions,
+                          collected_number NumAnomalousDimensions,
+                          collected_number AnomalyRate) {
+    static thread_local RRDSET *MLRS = nullptr;
+    static thread_local RRDDIM *NumTotalDimensionsRD = nullptr;
+    static thread_local RRDDIM *NumAnomalousDimensionsRD = nullptr;
+    static thread_local RRDDIM *AnomalyRateRD = nullptr;
+
+    if (!MLRS) {
+        MLRS = rrdset_create(
```

Comment:
please add user friendly descriptions into the `dashboard_info.js` to be shown on dashboard